packages/react-components/src/util/toFormattedBalance.ts

```diff
@@ -16,28 +17,23 @@ interface Arguments {
  * Examples (fixedPoint=2 and unit="Unit")
  * "123" ==> "1.23 Unit"
  */
-const toFormattedBalance = (args: Arguments): string => {
+const getFormattedBalance = (args: Arguments): string => {
   const DEFAULT_FIXED_POINT = 4;
   const DEFAULT_UNIT = '';
   const {
     balance,
     fixedPoint = DEFAULT_FIXED_POINT,
     unit = DEFAULT_UNIT
   } = args;
-  const balanceAsNumber = parseInt(balance, 10);
   const unitPart = unit ? ` ${unit}` : '';
 
-  if (!balance || isNaN(balanceAsNumber)) {
-    throw new Error(`Balance value is not valid: ${balance}`);
-  }
-
   /**
    * Condition 1: balance length is smaller than fixed point, e.g:
    * "123" ==> "0.1230" # when value length (3) is smaller than fixed point (4)
    */
-  if (balance.length <= fixedPoint) {
+  if (balance.toString().length <= fixedPoint) {
     const scalingSize = Math.pow(10, 1 - fixedPoint);
-    const valuePart = (balanceAsNumber * scalingSize).toFixed(fixedPoint);
+    const valuePart = (balance.toNumber() * scalingSize).toFixed(fixedPoint);
```

Comment:
A big number should remain big when doing mathematical operations. You shouldn't convert it to a number and then multiply it to scalingSize. You should  use mul or imul functions instead.

Reply:
This function is inside condition when balance value is smaller than 4, where I suppose we could consider the value as non-big number, which get the calculation straightforward. And the big number calculation is in the following condition in the traditional BN way.

Reply:
I would personally write it using the BN operation, so later if someone moves the code or changes the condition it proceeds to work as expected. However, this is acceptable.

Reply:
@alexsednz I just gave a quick try by using `BN divide method` which seems not work quite well for float - https://codesandbox.io/s/bnjs-e632o . 

If you do have any solutions, would you mind sharing here, otherwise, we could fix it later once we got a better solution?

Reply:
You are not doing any divide here, do you?

Reply:
`balance.toNumber() * scalingSize` is kind of dividing

Reply:
Also added multiple case in that demo just now, but seems not working as well